crates/router/src/core/payments/helpers.rs

```diff
@@ -2060,27 +2060,50 @@ pub fn generate_mandate(
 }
 
 // A function to manually authenticate the client secret with intent fulfillment time
-pub fn authenticate_client_secret(
+pub async fn authenticate_client_secret(
```

Comment:
do not make this function as async, let it just do the authentication and not any db calls. All the database calls can be made prior and data can be passed along.